src/view.js

```diff
@@ -301,6 +301,27 @@ const renderErrorDialog = (message, npub) => {
   return errorDialog;
 };
 
+export const init = async ({ npub, noteId, relays }) => {
+  let amountDialog
+  try {
+    if (!amountDialog) {
+      amountDialog = await renderAmountDialog({ npub, noteId, relays });
+    }
+    amountDialog.showModal();
+
+    if (!window.matchMedia("(max-height: 932px)").matches) {
+      amountDialog.querySelector('input[name="amount"]').focus();
+    }
```

Comment:
```suggestion
export const init = async ({ npub, noteId, relays, cachedAmountDialog }) => {
  let amountDialog = cachedAmountDialog;
  try {
    if (!amountDialog) {
      amountDialog = await renderAmountDialog({ npub, noteId, relays });
    }
    amountDialog.showModal();

    if (!window.matchMedia("(max-height: 932px)").matches) {
      amountDialog.querySelector('input[name="amount"]').focus();
    }

    return amountDialog;
```

this would allow you to cache the amount dialog and dry up the code a little. initTarget could be updated to:

```js
export const initTarget = (targetEl) => {
  let cachedAmountDialog = null;

  targetEl.addEventListener("click", async function () {
    const npub = targetEl.getAttribute("data-npub");
    const noteId = targetEl.getAttribute("data-note-id");
    const relays = targetEl.getAttribute("data-relays");

    cachedAmountDialog = await init({ npub, noteId, relays, cachedAmountDialog });
  });
};
```